File: dotnet/packages/Microsoft.TeamsAI/Microsoft.TeamsAI/Application/Authentication/TeamsSsoSettings.cs
using Microsoft.Identity.Client;

namespace Microsoft.Teams.AI
{
    /// <summary>
    /// Settings to initialize TeamsSsoAuthentication class
    /// </summary>
    public class TeamsSsoSettings
    {
        /// <summary>
        /// The AAD scopes for authentication. Only one resource is allowed in the scopes.
        /// </summary>
        public string[] Scopes { get; set; }

        /// <summary>
        /// The instance of ConfidentialClientApplication from Microsoft Authentication Library
        /// </summary>
        public IConfidentialClientApplication MSAL { get; set; }

        /// <summary>
        /// The sign in link for authentication.
        /// The library will pass `scope`, `clientId`, and `tenantId` to the link as query parameters.
        /// Your sign in page can leverage these parameters to compose the AAD sign-in URL.
        /// </summary>
        public string SignInLink { get; set; }

        /// <summary>
        /// Number of milliseconds to wait for the user to authenticate.
        /// Defaults to a value `900,000` (15 minutes).
        /// Only works in conversional bot scenario.
        /// </summary>
        public int Timeout { get; set; }

        /// <summary>
        /// Value indicating whether the authentication should end upon receiving an invalid message.
        /// Defaults to `true`.
        /// Only works in conversional bot scenario.
        /// </summary>
        public bool EndOnInvalidMessage { get; set; }

        /// <summary>
        /// Initializes the class
        /// </summary>
        /// <param name="scopes">The AAD scopes for authentication.</param>
        /// <param name="signInLink">The sign in link for authentication.</param>
        /// <param name="msal">The instance of ConfidentialClientApplication from Microsoft Authentication Library</param>
        /// <param name="timeout">Number of milliseconds to wait for the user to authenticate.</param>
        /// <param name="endOnInvalidMessage">Value indicating whether the authentication should end upon receiving an invalid message.</param>
        public TeamsSsoSettings(string[] scopes, string signInLink, IConfidentialClientApplication msal, int timeout = 900000, bool endOnInvalidMessage = true)
        {
            this.Scopes = scopes;
            this.MSAL = msal;
            this.SignInLink = signInLink;
            this.Timeout = timeout;
            this.EndOnInvalidMessage = endOnInvalidMessage;
        }
    }
}

File: js/packages/teams-ai/src/authentication/TeamsSsoSettings.ts
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

import { Configuration } from "@azure/msal-node";

/**
 * Represents the settings for Teams Single Sign-On (SSO) authentication.
 */
export interface TeamsSsoSettings {
    /**
     * The scopes required for authentication.
     */
    scopes: string[];
    /**
     * The MSAL (Microsoft Authentication Library) configuration.
     */
    msalConfig: Configuration;
    /**
     * The sign-in link for authentication.
     * The library will pass `scope`, `clientId`, and `tenantId` to the link as query parameters.
     * Your sign-in page can leverage these parameters to compose the AAD sign-in URL.
     */
    signInLink: string;
    /**
     * (Optional) number of milliseconds to wait for the user to authenticate.
     * Defaults to a value `900,000` (15 minutes.)
     * Only works in conversional bot scenario.
     */
    timeout?: number;
    /**
     * (Optional) value indicating whether the OAuthPrompt should end upon
     * receiving an invalid message. 
     * Defaults to `true`.
     */
    endOnInvalidMessage?: boolean;
}